Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -12,8 +12,56 @@
 
 namespace Web::FileAPI {
 
+// https://w3c.github.io/FileAPI/#convert-line-endings-to-native
+static ErrorOr<String> convert_line_endings_to_native(String const& string)
+{
+    // 1. Let native line ending be be the code point U+000A LF.
+    StringView native_line_ending = "\n"sv;
+
+    // FIXME: 2. If the underlying platform’s conventions are to represent newlines as a carriage return and line feed sequence, set native line ending to the code point U+000D CR followed by the code point U+000A LF.
+
+    // 3. Set result to the empty string.
+    StringBuilder result;
+
+    // 4. Let position be a position variable for s, initially pointing at the start of s.
+    u32 position = 0;
```

Comment:
Please use a `GenericLexer` for this, you can find a couple of examples in `Fetch/` :)

Reply:
Updated code looks :yakkiss:

Reply:
:yakblush: